subsys/fs/fs.c

```diff
@@ -0,0 +1,529 @@
+			return rc;
+		}
+	}
+
+	zdp->mp = NULL;
+	zdp->dops = NULL;
+	return rc;
+}
+
+/* Filesystem operations */
+int fs_mkdir(const char *abs_path)
+{
+	fs_mount_t *mp;
+	const char *fs_path;
+	int rc, match_len;
+
+	if ((strlen(abs_path) <= 1) || (abs_path[0] != '/')) {
+		SYS_LOG_ERR("invalid file name!!\n");
+		return -EINVAL;
+	}
+
+	rc = get_mnt_point(&mp, abs_path, &match_len);
+	if (rc < 0) {
+		SYS_LOG_ERR("%s:mount point not found!!\n", __func__);
+		return rc;
+	}
+	fs_path = &abs_path[match_len];
+
+	if (mp->fs->fs_ops->mkdir != NULL) {
+		rc = mp->fs->fs_ops->mkdir(mp, fs_path);
+		if (rc < 0) {
+			SYS_LOG_ERR("failed to create directory (%d)\n", rc);
+		}
+	}
+
+	return rc;
+}
+
+int fs_unlink(const char *abs_path)
+{
+	fs_mount_t *mp;
+	const char *fs_path;
+	int rc, match_len;
+
+	if ((strlen(abs_path) <= 1) || (abs_path[0] != '/')) {
+		SYS_LOG_ERR("invalid file name!!\n");
+		return -EINVAL;
+	}
+
+	rc = get_mnt_point(&mp, abs_path, &match_len);
+	if (rc < 0) {
+		SYS_LOG_ERR("%s:mount point not found!!\n", __func__);
+		return rc;
+	}
+	fs_path = &abs_path[match_len];
+
+	if (mp->fs->fs_ops->unlink != NULL) {
+		rc = mp->fs->fs_ops->unlink(mp, fs_path);
+		if (rc < 0) {
+			SYS_LOG_ERR("failed to unlink path (%d)\n", rc);
+		}
+	}
+
+	return rc;
+}
+
+int fs_stat(const char *abs_path, struct fs_dirent *entry)
+{
+	fs_mount_t *mp;
+	const char *fs_path;
+	int rc, match_len;
+
+	if ((strlen(abs_path) <= 1) || (abs_path[0] != '/')) {
+		SYS_LOG_ERR("invalid file name!!\n");
+		return -EINVAL;
+	}
+
+	rc = get_mnt_point(&mp, abs_path, &match_len);
+	if (rc < 0) {
+		SYS_LOG_ERR("%s:mount point not found!!\n", __func__);
+		return rc;
+	}
+	fs_path = &abs_path[match_len];
+
+	if (mp->fs->fs_ops->stat != NULL) {
+		rc = mp->fs->fs_ops->stat(mp, fs_path, entry);
+		if (rc < 0) {
+			SYS_LOG_ERR("failed get file or dir stat (%d)\n", rc);
+		}
+	}
+	return rc;
+}
+
+int fs_statvfs(const char *abs_path, struct fs_statvfs *stat)
+{
+	fs_mount_t *mp;
+	const char *fs_path;
+	int rc, match_len;
+
+	if ((strlen(abs_path) <= 1) || (abs_path[0] != '/')) {
+		SYS_LOG_ERR("invalid file name!!\n");
+		return -EINVAL;
+	}
+
+	rc = get_mnt_point(&mp, abs_path, &match_len);
+	if (rc < 0) {
+		SYS_LOG_ERR("%s:mount point not found!!\n", __func__);
+		return rc;
+	}
+	fs_path = &abs_path[match_len];
+
+	if (mp->fs->fs_ops->statvfs != NULL) {
+		rc = mp->fs->fs_ops->statvfs(mp, fs_path, stat);
+		if (rc < 0) {
+			SYS_LOG_ERR("failed get file or dir stat (%d)\n", rc);
+		}
+	}
+
+	return rc;
+}
+
+int fs_mount(fs_mount_t *mp)
+{
+	fs_mount_t *itr;
+	sys_dnode_t *node;
+	int rc = -EINVAL;
+
+	if ((mp == NULL) || (mp->mnt_point == NULL)) {
+		SYS_LOG_ERR("mount point not initialized!!\n");
+		return -EINVAL;
+	}
+
+	k_mutex_lock(&mutex, K_FOREVER);
+	/* check if requested file system is registered */
+	if (mp->type >= FS_TYPE_END ||  fs_map[mp->type] == NULL) {
+		SYS_LOG_ERR("requested FS not supported!!\n");
```

Comment:
Change log message: Not supported sounds like not supported at all, while fs type might be just not enabled in configuration.

Reply:
Ok will change the log message.